modules/mono/csharp_script.cpp

```diff
@@ -501,7 +501,8 @@ Vector<ScriptLanguage::StackInfo> CSharpLanguage::debug_get_current_stack_info()
 	_recursion_flag_ = true;
 	SCOPE_EXIT {
 		_recursion_flag_ = false;
-	};
+	}
+	FORCE_SEMICOLON
```

Comment:
Here's an example of the use of `FORCE_SEMICOLON`. Only use when the formatter is stubborn removing a necessary semi-colon.

Reply:
Alternatively:
```cpp
	SCOPE_EXIT { // clang-format off
		_recursion_flag_ = false;
	}; // clang-format on
```

Reply:
I don't know. Like, yeah it works, but what if a format error is introduced between them?

Reply:
If format errors in between are the concern, just move the `// clang-format off` down one line. Because the settings only apply to subsequent lines, this will turn clang-format off *only* for the line with the semicolon.
```cpp
	SCOPE_EXIT {
		_recursion_flag_ = false; // clang-format off
	}; // clang-format on
```
While that's the option I'd prefer most, here's two more alternatives: after looking at the `SCOPE_EXIT` macro, and this is the only place in the entire repo where it's used. I'd sooner say to either remove that macro entirely, or replace it with a version that wraps the content itself, both of which will fix clang-format removing the semicolon.
```cpp
	gdmono::ScopeExit GD_UNIQUE_NAME(gd_scope_exit) = gdmono::ScopeExitAux() + [=]() -> void {
		_recursion_flag_ = false;
	};
```
```cpp
#define SCOPE_EXIT(m_content) \
	auto GD_UNIQUE_NAME(gd_scope_exit) = gdmono::ScopeExitAux() + [=]() -> void { m_content };
//////////
	SCOPE_EXIT(_recursion_flag_ = false;)
```
---
I prefer these workarounds because I see `FORCE_SEMICOLON` better used for ensuring a trailing semicolon on macro. Here's an excerpt from one of my prospective PRs that wanted to tackle this on the clang-tidy scope:
```cpp
// By appending to a macro, ensures that it'll require a trailing semicolon.
#ifndef FORCE_SEMICOLON
#define FORCE_SEMICOLON static_assert(true)
#endif
```
`static_assert` produces zero build-time bloat, being a functionally "free" addition (especially in this always-true context). What it *does* provide is a way to require a trailing semicolon in virtually any context, as `static_assert` **requires** a semicolon to follow its declaration.